src/MatrixTypes_PETSc.f90

```diff
@@ -51,6 +51,9 @@ MODULE MatrixTypes_PETSc
 
   TYPE,EXTENDS(DistributedMatrixType) :: PETScMatrixType
     Mat :: A
+
+    !> Number of rows for nonsquare matrices:
+    INTEGER(SIK) :: m
```

Comment:
You call `m` The number of rows here, but you refer to it as the number of columns at lines 101 and 141.  Make those consistent (I assume columns is what you meant)

Reply:
Good catch, thanks!  I did mean columns there